### DISCUSSION TITLE
Delete app from teleport cluster

### DISCUSSION THREAD

**Discussion by mukerjee** [gravitational/teleport#12200] [Q&A] [answered]
Hello, I'm wondering if there's a way to manually delete an app from a teleport cluster. We are using fairly ephemeral machines and would like to not wait for the app / node to timeout (deleting a node manually works great, btw).

I can see from `tctl apps ls --format json` that apps do have `name`'s and `id`'s, but neither seem to work with `tctl rm apps/...`

e.g.,
```
tctl apps ls --format json
...
        "metadata": {
            "name": "mittens-infra1-vault",
            "expires": "2022-04-25T06:38:58Z",
            "id": 1650868138258650087
        },
...
$ tctl rm apps/1650868138258650087
ERROR: application "1650868138258650087" doesn't exist
$ tctl rm apps/mittens-infra1-vault
ERROR: application "mittens-infra1-vault" doesn't exist
```

Seems a bit weird since `tctl` recognizes that `apps/` exists, but just can't find the app i'm referring to.

**Comment by webvictim** [comment, marked as answer]
Try `tctl rm app/<name>` (as per https://goteleport.com/docs/application-access/guides/dynamic-registration/)

**Reply by mukerjee** [reply]
Great-- thanks for the help!

the issue was that I was registering apps statically (i.e., in the teleport.yaml on a node). it appears you can not delete static app registrations (which does make sense, now that i think about it more).

deleting dynamically registered apps works perfectly.
